VigneshDevHub/CampX
# Feature: Admin User Role for Enhanced Control

## 📋 Description
            This pull request introduces admin controls to enhance the management of user campgrounds within the CampX 
            application. It allows admins/maintainers to delete or edit any campground flagged as spam or inaccurate and delete 
            inaccurate reviews, ensuring better oversight and quality of content on the platform.

## 🔨 Changes Made
           - Fixed User model with role as user or admin.
           - Added functionality for identifying admin users in the system.
                  Admin should be the **first** to register for getting the role admin.
           - Admin can edit or delete any campground and delete reviews after login.
           - Users can edit or delete their own campgrounds and add reviews after register and login.

## ✅ Checklist
- ✅ My code follows the style guidelines of this project.
- ✅I have performed a self-review of my own code.
- ✅ I have commented my code, particularly in hard-to-understand areas.
- ✅ I have made corresponding changes to the documentation (if applicable).
- ✅ My changes generate no new warnings.
- ✅ I have tested it locally and it works fine.
- ✅ Any dependent changes have been merged and published in downstream modules.

## 🏷️ Types of Changes

What type of changes does your code introduce? (Check all that apply)

- ✅ New feature (non-breaking change which adds functionality) ✨
- [ ] UI enhancement (non-breaking change which enhances UI) 🎨
- ✅ Documentation update 📚

## 🤝 Related Issues
- This Fixes #5

## How to Test
     Register and login ### **first** as an admin user.
     Navigate to a campground or review created by a different user.
     Verify that you can edit or delete the campground and delete a review as intended.
     Test the spam management functionality.

Thankyou.
Regards 
Prerana Hippargi.
